### DISCUSSION TITLE
[infinity] All tables and data gone

### DISCUSSION THREAD

**Discussion by nainamd77** [infiniflow/infinity#800] [Q&A] [answered]
Hi,
We encountering below issues,
When we trying to start Infinity using below command, it is not starting.
"sudo systemctl start infinity"
So, we started it by just typing 'infinity' in the terminal prompt, which is running well along with log writings in terminal.

Another issue, 
If we restarted the system which hosting infinity, all the tables and its data gone. 
we need to recreate all again. 

please guide us to fix these issues.
infinity - version 0.1.0 release 2024-02-26

**Comment by JinHai-CN** [comment]
For the second issue, by default, the data directory of infinity is in /tmp, so every time the host restarts, /tmp will be cleared. The solution is to replace the /tmp directory with another directory. The data directory can be configured in conf/infinity_conf.toml. Then, you can use ./infinity -f conf/infinity_conf.toml to start the program.

**Reply by yuzhichang** [reply, marked as answer]
The rpm/deb usage instruction is https://github.com/infiniflow/infinity?tab=readme-ov-file#deploy-infinity-using-binary-package-on-linux-x86_64.

For example, on my OpenSUSE Tumbleweed:
```
zhichyu@germ:~/Downloads$ sudo rpm -Uvh --force infinity-0.1.0-dev-x86_64.rpm
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
Updating / installing...
1:infinity-0.1.0_dev-1             ################################# [ 50%]
warning: /usr/lib/systemd/system/infinity.service saved as /usr/lib/systemd/system/infinity.service.rpmsave
Cleaning up / removing...
2:infinity-0.1.0_dev1-1            ################################# [100%]
zhichyu@germ:~/Downloads$ sudo systemctl start infinity
zhichyu@germ:~/Downloads$ sudo systemctl status infinity
● infinity.service - Infinity db
Loaded: loaded (/usr/lib/systemd/system/infinity.service; disabled; preset: disabled)
Active: active (running) since Sat 2024-03-16 20:25:27 CST; 3s ago
Main PID: 5531 (infinity)
Tasks: 152 (limit: 4915)
CPU: 28ms
CGroup: /system.slice/infinity.service
└─5531 /usr/bin/infinity -f /usr/etc/infinity_conf.toml

Mar 16 20:25:27 germ infinity[5531]:  - full_checkpoint_txn_interval: 0
Mar 16 20:25:27 germ infinity[5531]:  - delta_checkpoint_interval_sec: 60
Mar 16 20:25:27 germ infinity[5531]:  - delta_checkpoint_interval_wal_bytes: 1000000000
Mar 16 20:25:27 germ infinity[5531]:  - wal_size_threshold: 1.00GB
Mar 16 20:25:27 germ infinity[5531]:  - wal_dir: /var/infinity/wal
Mar 16 20:25:27 germ infinity[5531]:  - flush_at_commit: OnlyWrite
Mar 16 20:25:27 germ infinity[5531]:  - dictionary_dir: /var/infinity/resource
Mar 16 20:25:27 germ infinity[5531]: Release: 0.1.0 build on 2024-03-16 00:48.16 with Release mode from branch: main, commit-id: e85d19248fb365c0b039dfcc57c48b17d982f409
Mar 16 20:25:27 germ infinity[5531]: HTTP server listen on port: 23820
Mar 16 20:25:27 germ infinity[5531]: API server listen on: 0.0.0.0:23817, thread pool: 128
```

The service unit file is `/usr/lib/systemd/system/infinity.service`. 
The config file is `/usr/etc/infinity_conf.toml`.
The config file says that the main directory is `/var/infinity` so that system reboot doesn't cause data loss.

`infinify` database is a long running service. So we recommend rpm or deb package to run it as a systemd service if you have root permission of the host. If you don't have the permission, use tgz package  to run it under a common user.

**Reply by nainamd77** [reply]
we uninstalled infinity and removed all related files, then re-installed with latest release. Now working well it started with system and without data loss.
